### ISSUE TITLE
events: search and filters

### ISSUE THREAD

**Issue by BrunoRosendo** [NIAEFEUP/website-niaefeup-frontend#15]
depends on #14 

Following the work already done on the events page and the mockups (screenshot below), make the search bar and displayed filters functional.

Since the number of events should be reduced (and the backend does not support filters anyway), I believe it is more advantageous for this feature to be fully implemented in the frontend by updating the events list whenever the user clicks on a filter or types/submits in the text bar.

![image](https://user-images.githubusercontent.com/55768934/210679542-a4df5d85-819b-4e47-8879-d3d5a0eb915d.png)
